OpenSim/Simulation/Model/AbstractTool.cpp

```diff
@@ -553,22 +553,31 @@ printResults(const string &aBaseName,const string &aDir,double aDT,
 }
 
 
-bool AbstractTool::createExternalLoads( const string& aExternalLoadsFileName, Model& aModel, const Storage *loadKinematics)
+bool AbstractTool::createExternalLoads( const string& aExternalLoadsFileName,
+                                        Model& aModel, const Storage *loadKinematics)
 {
     if(aExternalLoadsFileName==""||aExternalLoadsFileName=="Unassigned") {
         cout<<"No external loads will be applied (external loads file not specified)."<<endl;
         return false;
     }
 
+    Model copyModel = aModel;
+    // speedup realize position calculations by removing all force elements
```

Comment:
I'm not sure I understand the downstream effects of this change, instead of the ExternalLoads added to passed in Model as suggested by the signature, now a copy of the Model is made and the  passed in Model is unchanged, what's the rationale?

Reply:
I separate to things that were going on with the passed in model.
1. It was being used to compute the transformation of the force into the body in which it was being applied.
2. It was being augmented with ExternalLoads.

Doing both these operations on the same Model means you have to remove the first ExternalLoads that was added for the purpose of computing transformed points.

Since we only really care about getting the passed in model back with the correct ExternalLoads, it seems safer and more straightforward to use a copy to be the workhorse for doing the transformation. In that case, we might as strip it down to its bare bones to do the transformation as quickly as possible. Forces are completely unnecessary for the realization of position transforms.

Once we are happy with the ExternalLoads we then add it the model we care about that has remained untouched.